frontend/service/Ethers.ts

```diff
@@ -89,6 +90,65 @@ const checkRpc = async (rpc: string): Promise<boolean> => {
   }
 };
 
+const BACK_TRACK_BLOCKS = 9000;
+const MAX_ROUNDS = 10;
+
+/**
+ * Get the latest transaction details for the given contract address
+ */
+export const getLatestTransaction = async (
+  rpc: string,
+  contractAddress: Address,
+): Promise<TransactionInfo | null> => {
+  const provider = new providers.StaticJsonRpcProvider(rpc, {
```

Comment:
can reuse the provider from `constants/providers.ts` (probably belongs in `utils/...` if i'm honest)